pkg/auth/handler/login.go

```diff
@@ -72,6 +72,12 @@ type LoginHandler struct {
 	AuthInfoStore        authinfo.Store              `dependency:"AuthInfoStore"`
 	PasswordAuthProvider password.Provider           `dependency:"PasswordAuthProvider"`
 	UserProfileStore     dependency.UserProfileStore `dependency:"UserProfileStore,optional"`
+	TxContext            db.TxContext                `dependency:"TxContext"`
+}
+
+// WithTx describes if the api handler requires transaction
+func (h LoginHandler) WithTx() bool {
+	return true
 }
```

Comment:
Let don't define this at every handler as dependency?

I prefer this optional at the respective Factory.NewHandler.

Reply:
I actually prefer having this function explicitly.

If later someone want to figure out whether a handler is wrapped inside a transaction, this function can tell immediately.